modules/php/Cards/Rules/RuleGetOnWithIt.php

```diff
@@ -9,22 +9,53 @@ public function __construct($cardId, $uniqueId)
   {
     parent::__construct($cardId, $uniqueId);
 
-    $canBeUsedByPlayer = true;
-
     $this->name = clienttranslate("Get On With It!");
     $this->subtitle = clienttranslate("Free Action");
     $this->description = clienttranslate(
       "Before your final play, if you are not empty handed, you may discard your entire hand and draw 3 cards. Your turn then ends immediately."
     );
   }
 
+  public function canBeUsedInPlayerTurn($player_id)
+  {
+    $game = Utils::getGame();
+    return $game->cards->countCardInLocation("hand", $player_id) > 0
+      && Utils::calculateCardsLeftToPlayFor($player_id) > 0;
+  }
+  
   public function immediateEffectOnPlay($player)
   {
-    // @TODO
+    // nothing
   }
 
   public function immediateEffectOnDiscard($player)
   {
     // nothing
   }
+
+  public function freePlayInPlayerTurn($player_id)
+  {
+    $game = Utils::getGame();
+    // Discard entire hand
+    $cards = $game->cards->getCardsInLocation("hand", $player_id);
+    foreach ($cards as $card_id => $card) {
+      $game->cards->playCard($card_id);
+    }
+
+    $game->notifyAllPlayers("handDiscarded", "", [
+      "player_id" => $player_id,
+      "cards" => $cards,
+      "discardCount" => $game->cards->countCardInLocation("discard"),
+      "handCount" => $game->cards->countCardInLocation("hand", $player_id),
+    ]);    
+    // Draw 3 cards (+ inflation)    
+    $addInflation = Utils::getActiveInflation() ? 1 : 0;
+    $drawCount = 3 + $addInflation;
+    $game->performDrawCards($player_id, $drawCount);
+
+    // Force end of turn (set count cards played above 999)
+    $game->setGameStateValue("playedCards", PLAY_COUNT_ALL + 1);
```

Comment:
Not a big fan of this one. Could we return the transition instead of `null` like we do for actions?

Reply:
Hmm, my reasoning was to avoid having to repeat any kind of end-turn checks or code here.
But you are right, that would be much cleaner if it was possible by just transitioning via "endOfTurn" to next state.
Because now I actually had to hack a check for PLAY_COUNT_ALL in PlayCards trait also.

To be investigated further if state transition here doesn't have any other unwanted side effects.

Reply:
You were absolutely right here, simplified everything via "endOfTurn" state transition